### PR TITLE
Implement Collect method 

### PR DESCRIPTION
**Pull request by katejefferson**
https://dev.recurly.com/v2.5/docs/collect-an-invoice

### CODE REVIEW

File: services.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ type InvoicesService interface {
 	GetPDF(invoiceNumber int, language string) (*Response, *bytes.Buffer, error)
 	Preview(accountCode string) (*Response, *Invoice, error)
 	Create(accountCode string, invoice Invoice) (*Response, *Invoice, error)
+	Collect(invoiceNumber int) (*Response, *Invoice, error)
```
Comment (cristiangraz) [inline services.go]: Could you add this to `mock.InvoicesService` as well?

File: invoices_service.go [hunk 2]
```diff
@@ -140,6 +140,25 @@ func (s *invoicesImpl) Create(accountCode string, invoice Invoice) (*Response, *
...
+// https://dev.recurly.com/v2.5/docs/collect-an-invoice
+func (s *invoicesImpl) Collect(invoiceNumber int) (*Response, *Invoice, error) {
+	action := fmt.Sprintf("invoices/%d/collect", invoiceNumber)
+	req, err := s.client.newRequest("PUT", action, nil, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var dst Invoice
+	resp, err := s.client.do(req, &dst)
```
Comment (cristiangraz) [inline invoices_service.go]: Could you return a `nil` invoice and add a corresponding test when the invoice isn't retrieved? See this PR for reference: https://github.com/blacklightcms/recurly/pull/19

File: services.go (hunk 1 above)
Comment (katejefferson) [inline services.go]: Added in [e7ca00](https://github.com/blacklightcms/recurly/pull/21/commits/fcbd89a3b82ff5a45a6e012e560a106c60e7ca00).

File: invoices_service.go (hunk 2 above)
Comment (katejefferson) [inline invoices_service.go]: Added in [e7ca00](https://github.com/blacklightcms/recurly/pull/21/commits/fcbd89a3b82ff5a45a6e012e560a106c60e7ca00).

**Pull request merged**